.github/workflows/validate.yml

```diff
@@ -157,21 +157,24 @@ jobs:
       contents: read
       pull-requests: read
     runs-on: ubuntu-24.04
-    # Only check commits on pull requests.
-    if: github.event_name == 'pull_request'
     steps:
       - name: get pr commits
+        if: github.event_name == 'pull_request' # Only check commits on pull requests.
         id: 'get-pr-commits'
         uses: tim-actions/get-pr-commits@v1.3.1
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
 
       - name: check subject line length
+        if: github.event_name == 'pull_request' # Only check commits on pull requests.
         uses: tim-actions/commit-message-checker-with-regex@v0.3.2
         with:
           commits: ${{ steps.get-pr-commits.outputs.commits }}
           pattern: '^.{0,72}(\n.*)*$'
           error: 'Subject too long (max 72)'
+      - name: succeed (not a PR)
+        if: github.event_name != 'pull_request'
+        run: echo "Nothing to check here."
```

Comment:
I guess this is needed because _something_ inside the commit check thing needs to run to be marked as succeeded? In that case, can you add a comment explaining this?

Reply:
First, I don't like this kludge myself, and so I spent some time to find a better way to do that (such as, adding `commit` job as a `needs:` to `all-done` conditionally), but nothing works. If there's a way that I missed, let me know.

Second, this code is added by a separate commit, with a decent explanation which can be found via git blame.

Here's a copy-paste from the commit message as it is:

>    gha/ci: allow validate/all-done to succeed for non-PRs
>    
>    When we run CI not on a pull request, the commit job is skipped, as a
>    result, all-done is also skipped.
>    
>    To allow all-done to succeed, modify the commit job to succeed for
>    non-PRs.

I've added another short comment to the `.yml`